mlflow/sklearn.py

```diff
@@ -30,13 +28,16 @@
 
 FLAVOR_NAME = "sklearn"
 
-DEFAULT_CONDA_ENV = _mlflow_conda_env(
-    additional_conda_deps=[
-        "scikit-learn={}".format(sklearn.__version__),
-    ],
-    additional_pip_deps=None,
-    additional_conda_channels=None,
-)
+
+def get_default_conda_env():
```

Comment:
Other flavor definitions use top-level imports for dependencies that are not installed by MLflow directly (e.g., `mlflow.tensorflow`) and define a module-level `DEFAULT_CONDA_ENV` variable. For consistency, we should update the other flavors or continue to use top-level imports (afaik, we haven't received complaints about the current paradigm).

Reply:
Hmm, the problem with top-level imports is that if any internal code imports the module in order to use any other methods or variables, it will fail at module load time. For example, how the sagemaker code uses mleap -- it doesn't always need it, but it would suddenly require scikit-learn if it were top-level.

Reply:
That's fair. I'll update the other flavor modules accordingly!